dll/win32/browseui/shellbars/CBandSiteMenu.cpp

```diff
@@ -33,14 +44,37 @@ CBandSiteMenu::~CBandSiteMenu()
     if (m_hmenu)
         DestroyMenu(m_hmenu);
 
-    if (m_menuDsa)
-        DSA_Destroy(m_menuDsa);
+    if (m_comcatDsa)
```

Comment:
Maybe a custom `CHeapPtr` with a DSA allocator? (See `CComAllocator`)

Reply:
Yeah that would help a lot with freeing pild. I'm not sure that this is applicable to DSAs though (we could just use an ATL list).